pydantic/datetime_parse.py

```diff
@@ -143,7 +143,23 @@ def parse_time(value: Union[time, StrBytesIntFloat]) -> time:
     if kw['microsecond']:
         kw['microsecond'] = kw['microsecond'].ljust(6, '0')
 
-    kw_ = {k: int(v) for k, v in kw.items() if v is not None}
+    tzinfo_str = kw.pop('tzinfo')
+    if tzinfo_str == 'Z':
```

Comment:
looks like this is duplicated from `parse_datetime`, maybe extract to a new file.

Reply:
I think you meant to write function instead of file? - as updated?

Reply:
In moving this code to another function I spotted that the `datetime` parsing is also able to raise a ValueError exception when a bad TZ is given. There is an new test case demonstrating this which is fixed by the last commit, instead raising a `DateTimeError`.